### PR TITLE
refactor: format files with uncrustify

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/mbyte.c [hunk 1]
```diff
@@ -353,12 +356,15 @@ int enc_canon_props(const char_u *name)
   int i;
 
   i = enc_canon_search(name);
-  if (i >= 0)
+  if (i >= 0) {
     return enc_canon_table[i].prop;
-  if (STRNCMP(name, "2byte-", 6) == 0)
+  }
```
Comment (bfredl) [inline src/nvim/mbyte.c:361]: I'd add an else here manually to get rid of extra line, and same three lines below.

File: src/nvim/search.c [hunk 2]
```diff
@@ -998,41 +1009,37 @@ static int first_submatch(regmmatch_T *rp)
...
+/// Careful: If spats[0].off.line == TRUE and spats[0].off.off == 0 this
+/// makes the movement linewise without moving the match position.
+///
+/// @param oap           can be NULL
+/// @param dirc          '/' or '?'
+/// @param search_delim  delimiter for search, e.g. '%' in s%regex%replacement
+/// @param sia           optional arguments or NULL
+///
+/// @return              0 for failure, 1 for found, 2 for found and line offset added.
+int do_search(oparg_T         *oap, int dirc, int search_delim, char_u          *pat, long count,
```
Comment (bfredl) [inline src/nvim/search.c:1036]: I suppose re-running uncrustify will fix this?

File: src/nvim/search.c [hunk 3]
```diff
@@ -998,41 +1009,37 @@ static int first_submatch(regmmatch_T *rp)
...
-    int dirc,                       // '/' or '?'
-    int search_delim,  // delimiter for search, e.g. '%' in s%regex%replacement
-    char_u          *pat,
-    long count,
-    int options,
-    searchit_arg_T  *sia        // optional arguments or NULL
-)
+/// Highest level string search function.
+/// Search for the 'count'th occurrence of pattern 'pat' in direction 'dirc'
+///    If 'dirc' is 0: use previous dir.
```
Comment (bfredl) [inline src/nvim/search.c:1014]: these could be made into `@param`:s while we're at it.

File: src/nvim/search.c (hunk 2 above)
Comment (dundargoc) [reply src/nvim/search.c]: Nah, there are a bunch of options specifically for asterisk that I've ignored until now. I've fixed this but not sure if the fix unfixed something else.

**Comment by dundargoc** [comment]
@bfredl @justinmk I've fixed the review comments, feel free to take a look whenever you have the time.

**Review by justinmk — APPROVED** [review]
LGTM

**Pull request merged**